components/shared/base/cross_process_instant.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2024 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! In implementation a monotonic nanosecond precision timer, like [`std::time::Instant`]
+//! that can be serialized and compared across processes.
+
+use std::ops::{Add, Sub};
+
+use malloc_size_of_derive::MallocSizeOf;
+use serde::{Deserialize, Serialize};
+use time_03::Duration;
+
+/// A monotonic, nanosecond precision timer that can be used cross-process. The value
+/// stored internally is purposefully opaque as the origin is platform-specific. They can
+/// be compared and [`time_03::Duration`] can be found by subtracting one from another.
+/// The `time` crate is used in this case instead of `std::time` so that durations can
+/// be negative.
+#[derive(
+    Clone, Copy, Debug, Deserialize, Eq, MallocSizeOf, Ord, PartialEq, PartialOrd, Serialize,
+)]
+pub struct CrossProcessInstant {
+    value: u64,
+}
+
+impl CrossProcessInstant {
+    pub fn now() -> Self {
+        Self {
+            value: platform::now(),
+        }
+    }
+
+    /// Some unspecified time epoch. This is mainly useful for converting
+    /// [`CrossProcessInstant`] into a `DOMHighResolutionTimestamp`.
+    ///
+    /// TODO: This should be replaced with a proper implementaiton of `timeOrigin`. See
+    /// <https://html.spec.whatwg.org/multipage/#concept-settings-object-time-origin>
```

Comment:
The more precise spec link would be https://w3c.github.io/hr-time/#sec-time-origin.

Reply:
I've updated the spec link and improve the comment a bit to more accurately reflect the state of the code.